great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+from typing import Dict, List, Optional, Union
+
+from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
+from great_expectations.exceptions import ProfilerConfigurationError
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder import DomainBuilder
+from great_expectations.rule_based_profiler.domain_builder.cardinality_checker import (
+    CardinalityChecker,
+    CardinalityLimitMode,
+)
+from great_expectations.rule_based_profiler.domain_builder.domain_builder import (
+    build_domains_from_column_names,
+)
+from great_expectations.rule_based_profiler.types import Domain, ParameterContainer
+from great_expectations.validator.metric_configuration import MetricConfiguration
+from great_expectations.validator.validator import Validator
+
+
+class CategoricalColumnDomainBuilder(DomainBuilder):
```

Comment:
@anthonyburdi Can this domain builder allow columns to be set on it just like `ColumnDomainBuilder`?

Reply:
Not currently, checking with Eugene on requirements. Will report back.

Reply:
Confirmed that if a user wants to specify specific columns they would use a ColumnDomainBuilder specifying those specific columns in a different rule.